cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -216,8 +217,11 @@ private PluginProperties substituteMacroFields(Plugin plugin, MacroEvaluator mac
     Map<String, PluginPropertyField> pluginPropertyFieldMap = plugin.getPluginClass().getProperties();
     MacroParser macroParser = new MacroParser(macroEvaluator);
     for (Map.Entry<String, PluginPropertyField> pluginEntry : pluginPropertyFieldMap.entrySet()) {
+      String macroValue = plugin.getProperties().getProperties().get(pluginEntry.getKey());
+      if (Strings.isNullOrEmpty(macroValue)) {
```

Comment:
And if the field is not macro supported, you still skip it? Seems not right to me.